### PR TITLE
source-google-calendar 

### LINKED ISSUE #45995: [connector-request] Google Calendar
This is a **suggestion** to create a connector for **Google Calendar** using the UI Connector Builder.
The website for the connector is https://calendar.google.com.

If you want to try, claim this issue and start working on it following the steps below.

### Steps:
1. Comment in the issue and wait to be assigned to start working on it.
2. Map the API endpoints (get approval to move to 3 step)
3. Describe steps to get credentials.
4. Create the connector using the UI Builder

> [!IMPORTANT]
> Airbyte doesn't have sandbox credentials for this connector. You must have access/credentials to the service provider to create the connector.

> [!CAUTION]
> This is a **suggestion** and there may be cases where creating the connector using the Builder won't be possible. To minimize the risk of investing a lot of time directly in creating the connector, we strongly recommend to follow steps above.

### PR DESCRIPTION
**Pull request by bala-ceg**
## What

This PR adds a new connector for source Google Calendar (source-google-calendar).

Contributor-provided connector description:

Solves https://github.com/airbytehq/airbyte/issues/45995
## Details

<details><summary>Expand for list of spec properties and streams</summary>

### Configuration

| Input | Type | Description | Default Value |
|-------|------|-------------|---------------|
| `client_id` | `string` | Client ID.  |  |
| `client_secret` | `string` | Client secret.  |  |
| `client_refresh_token_2` | `string` | Refresh token.  |  |
| `calendarid` | `string` | Calendar Id.  |  |

### Streams
| Stream Name | Primary Key | Pagination | Supports Full Sync | Supports Incremental |
|-------------|-------------|------------|---------------------|----------------------|
| colors | calendar.event | No pagination | ✅ |  ❌  |
| settings | id | DefaultPaginator | ✅ |  ❌  |
| calendarlist | id | DefaultPaginator | ✅ |  ❌  |
| calendars | id | DefaultPaginator | ✅ |  ❌  |
| events | id | DefaultPaginator | ✅ |  ❌  |

</details>

## Reviewer checklist
- [ ] Run `/format-fix` to resolve any formatting errors
- [ ] Help the contributor upload an icon that follows our icon design guidelines
- [ ] Click into the CI workflows that wait for a maintainer to run them, which should trigger CI runs

### CODE REVIEW

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Left some comments.

File: docs/integrations/sources/google-calendar.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Google Calendar
+Solves https://github.com/airbytehq/airbyte/issues/45995
```
[the remark this answers is not in the document]
Comment (bala-ceg) [reply docs/integrations/sources/google-calendar.md:2]: Updated the Readmed.md

File: docs/integrations/sources/google-calendar.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+| `calendarid` | `string` | Calendar Id.  |  |
+
+## Streams
+| Stream Name | Primary Key | Pagination | Supports Full Sync | Supports Incremental |
+|-------------|-------------|------------|---------------------|----------------------|
+| colors |  | No pagination | ✅ |  ❌  |
+| settings | id | DefaultPaginator | ✅ |  ❌  |
+| calendarlist | id | DefaultPaginator | ✅ |  ❌  |
+| calendars | id | DefaultPaginator | ✅ |  ❌  |
+| events |  | DefaultPaginator | ✅ |  ❌  |
```
[the remark this answers is not in the document]
Comment (bala-ceg) [reply docs/integrations/sources/google-calendar.md]: Thanks @marcosmarxm. I updated the stream.

**Review by marcosmarxm — APPROVED** [review]
LGTM Awesome contribution!

**Pull request merged**